tests/array/array-sortStable.dora

```diff
@@ -1,45 +1,44 @@
-//= ignore
-import std::Assert;
-
 fun main() {
-  test1();
-  test2();
-  test3();
+  int32();
+  int64();
+  float64();
 }
 
-fun test1() {
+fun int32() {
   let x = Array[Int32](2, 3, 0, 4);
   Array[Int32]::sortStable(x);
   assert(x == Array[Int32](0, 2, 3, 4));
 }
 
-fun test2() {
-  let x = Array[Int32](5, 1, 4, 2, 8);
-  Array[Int32]::sortStable(x);
-  assert(x == Array[Int32](1, 2, 4, 5, 8));
+fun int64() {
+  let x = Array[Int64](5L, 1L, 4L, 2L, 8L);
+  Array[Int64]::sortStable(x);
+  assert(x == Array[Int64](1L, 2L, 4L, 5L, 8L));
 }
 
-// this is not entirely correct, as pNaN and nNaN positions' should be swapped.
-// this is an issue of the `sortsAs` implementation, not of the `sortStable` method.
-fun test3() {
+fun float64() {
+  // as x86 returns a negative NaN for 0.0/0.0, but Arm&RiscV return a positive NaN,
+  // we need to check first to make sure the NaN we are using has the desired sign.
+  let isX86 = (0.0/0.0).abs() !== 0.0/0.0;
+
   // expected order
-  let nNaN = -(0.0/0.0);
+  let nNaN = if isX86 { 0.0/0.0 } else { -(0.0/0.0) };
```

Comment:
Do we need `isX86` at all? Couldn't we build nNaN with `-(0.0/0.0).abs()` and pNaN with `(0.0/0.0).abs()`. IMHO this would be a bit clearer to me.

Reply:
I did it that way to make it a bit more obvious that this is related to platform-dependent behavior, but I can change it if you like.

Reply:
Yeah, I would prefer that. We have that comment of yours to explain why we do this...

Reply:
Ok, will do1